### ISSUE TITLE
http status code when accessing someone else's (private) database should be 404

### ISSUE THREAD

**Issue by justinclift** [sqlitebrowser/dbhub.io#66]
At the moment it's giving "internal server error" (5xx), instead of "not found" due to the access failure being communicated poorly do the function which calls it.

This should be addressed so that it's not possible to figure out people's private database names by checking the difference in return codes.

**Comment by chrisjlocke** [comment]
Sort of linked to this (maybe?) shouldn't there be some security layer which says, "ok, I've just given you 5 (10?) 404 error codes now.  You're clearly fishing so have this old boot.  Bugger off." and then just block them?  I know from my Wordpress hosting sites, people will attempt to guess the passwords for users (most Wordpress sites have an 'admin' user... unless you change it) and more annoyingly, spoof their IPs, to give them more chances...

**Comment by justinclift** [comment]
Yeah, we could definitely add some kind of counter in the background for various weird things (eg detect likely abuse), and block the ones [for some period of time] which cross some threshold.

Not sure if we need to implement it for the authentication layer though.  In theory Auth0 already has detection for brute forcing attempts (and similar), though I've not actually looked closely.

Actually, having seen how Auth0 responds to other problems and major bugs in their platform (very, very slowly), I'd not be inclined to put much faith in them being on top of things. ☹️ 

So yeah, maybe some kind of implementation for even the auth layer would make sense. :smile:

**Comment by chrisjlocke** [comment]
> Yeah, we could definitely add some kind of counter in the background for various weird things

Shall I add that as a new issue, or do you want to keep this one open?

**Comment by justinclift** [comment]
Make a new issue for it. :smile:
